addons/sale/models/sale_order_line.py

```diff
@@ -560,14 +560,20 @@ def _compute_pricelist_item_id(self):
 
     @api.depends('product_id', 'product_uom', 'product_uom_qty')
     def _compute_price_unit(self):
+        def has_manual_price(line):
+            # `line.currency_id` can be False for NewId records
+            currency = line.currency_id or line.env.company.currency_id
```

Comment:
Considering `_compute_amount` falls back on `or self.order_id.company_id.currency_id` instead (see `_prepare_base_line_for_taxes_computation`), we probably could use the same fallback here ? Or at least, it could take precedence on the environment company currency.

```suggestion
            currency = (
                line.currency_id
                or line.order_id.company_id.currency_id
                or line.env.company.currency_id
            )
```
would make sense when the main company of the environment is different than the order company